src/backend/app-core/middleware.go

```diff
@@ -50,6 +50,37 @@ func (p *portalProxy) sessionMiddleware(h echo.HandlerFunc) echo.HandlerFunc {
 	}
 }
 
+// Support for Angular XSRF
+func (p *portalProxy) xsrfMiddleware(h echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		log.Debug("xsrfMiddleware")
+
+		// Only do this for mutating requests - i.e. we can ignore for GET or HEAD requests
+		if c.Request().Method() == "GET" || c.Request().Method() == "HEAD" {
+			return h(c)
+		}
+		errMsg := "Failed to get stored XSRF token from user session"
```

Comment:
It's recommended to check for cross-origin requests at this point using request headers: https://www.owasp.org/index.php/Cross-Site_Request_Forgery_(CSRF)_Prevention_Cheat_Sheet#Verifying_Same_Origin_with_Standard_Headers.